### PR TITLE
Added Window.Open and window.SetLocation methods for the Window struct

### PR DESCRIPTION
**Pull request by aelango97**
Added 2 new methods:
 + `window.open` : https://developer.mozilla.org/en-US/docs/Web/API/Window/open#Syntax
 + `window.location = someURL`

### CODE REVIEW

File: window.go [hunk 1]
```diff
@@ -26,6 +26,14 @@ func (w *Window) AddEventListener(typ string, h EventHandler) {
 	}))
 }
 
+func (w *Window) Open(url, windowName, windowFeatures string) {
+	w.v.Call("open", url, windowName, windowFeatures)
```
Comment (dennwc) [inline window.go]: As a though, the `windowFeatures` can be a `map[string]string`, and the function can convert it to `k1=v1,k2=v2`.

What do you think?

Comment (aelango97) [reply window.go]: actually yeah that makes sense.

Comment (aelango97) [reply window.go]: but what about the string escaping?

Comment (aelango97) [reply window.go]: @dennwc I have added the same as per your suggestion, can you review it 😄

File: window.go [hunk 2]
```diff
@@ -26,6 +30,26 @@ func (w *Window) AddEventListener(typ string, h EventHandler) {
 	}))
 }
 
+func (w *Window) Open(url, windowName, windowFeatures map[string]string) {
```
Comment (dennwc) [inline window.go]:
```suggestion
func (w *Window) Open(url, windowName string, windowFeatures map[string]string) {
```

File: window.go [hunk 3]
```diff
@@ -26,6 +30,26 @@ func (w *Window) AddEventListener(typ string, h EventHandler) {
...
 	w.AddEventListener("resize", fnc)
 }
+
+func joinKeyValuePairs(kvpair map[string]string, joiner string) (ret string) {
+	if kvpair == nil {
+		return ""
+	}
+
+	for k, v := range kvpair {
+		ret += k + "=" + v + joiner
```
Comment (dennwc) [inline window.go]: It's better to do this via `strings.Join`

File: window.go (hunk 1 above)
Comment (dennwc) [reply window.go]: not sure about string escaping, we will need to check it separately

File: window.go (hunk 3 above)
Comment (aelango97) [reply window.go]: will do

Comment (aelango97) [reply window.go]: `strings.Join` is for string list right?

File: window.go (hunk 2 above)
Comment (aelango97) [reply window.go]: whoops

**Comment by aelango97** [comment]
@dennwc  done some changes..

**Pull request merged**